src/init.cpp

```diff
@@ -1976,9 +1974,18 @@ bool AppInit2(boost::thread_group& threadGroup, CScheduler& scheduler)
 
     connOptions.nLocalServices      = nLocalServices;
     connOptions.nMaxConnections     = nMaxConnections;
-    connOptions.nSendBufferMaxSize  = 1000*GetArg("-maxsendbuffer", 1*1000);
-    connOptions.nReceiveFloodSize   = 1000*GetArg("-maxreceivebuffer", 5*1000);
-
+    unsigned int fromKBtoB = 1000; // from kilobytes to bytes
+    int64_t msb_unbound = GetArg("-maxsendbuffer", DEFAULT_MAX_SEND_BUFFER);
+    if (msb_unbound > 0 && msb_unbound <= std::numeric_limits<unsigned int>::max() / fromKBtoB)
```

Comment:
I would personally enforce a more strict range of values. While this check solves the underflow issue we must consider this is a limit per connection and the way it is checked it allows up to 2Gigabytes unprocessed messages per socket.
I understand this is a condition unlikely to happen but ... as always, better safe than sorry.
Maybe a hard limit of, say, 50MeBi is appropriate ?

Reply:
Acknowledged. Let me add some comments here:

> and that users who change that value are supposed to be experts

this IMHO should not be assumed: on the contrary we should be providing a software solution that is "stupid proof" to help spread the node instances even for "curious". If instead the aim is to address mainly professional node operators then I cannot disagree but it seems it goes agains a decentralization ethos

> it's hard to find a proper max limit and it seems a little subjective (why 50 MB? why not 100 or higher?)

Given the number of max connections allowed an overall allotment of "unprocessed messages" is IMHO advisable: it helps identify why we may get in such an unfortunate solution

> a max limit that makes sense today may be too low in the next years and would require continuous adjustments to the code

I would expect the code to be continuosly adjusted in the next years: we should not be afraid of the workload required to continuosly improve and adjust the UX.

Last but not least : actual implementation might allow the insertion of a value as low as 1 which results in a 1'000 byte "spamming threshold". In such a condition a node sending a full block of 2MB would be deemed as spammer and disconnected. At the very least I would enforce a value which is GE than max message size.